modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/AbstractDefaultSchemaTest.java

```diff
@@ -0,0 +1,140 @@
+            prefix += "PUBLIC.";
+
+        return prefix + TBL_NAME;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        startGrid(0);
+    }
+
+    /** */
+    @Test
+    public void testBasicOpsExplicitPublicSchema() {
+        executeStmtsAndVerify(() -> true);
+    }
+
+    /** */
+    @Test
+    public void testBasicOpsImplicitPublicSchema() {
+        executeStmtsAndVerify(() -> false);
+    }
+
+    /** */
+    @Test
+    public void testBasicOpsMixedPublicSchema() {
+        AtomicInteger i = new AtomicInteger();
+
+        executeStmtsAndVerify(() -> (i.incrementAndGet() & 1) == 0);
+    }
+
+    /** */
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testCreateDropNonExistingSchema() {
+        GridTestUtils.assertThrowsWithCause(
+            () -> sql("CREATE TABLE UNKNOWN_SCHEMA." + TBL_NAME + "(id INT PRIMARY KEY, val INT)"),
+            SQLException.class
+        );
+
+        GridTestUtils.assertThrowsWithCause(
+            () -> sql("DROP TABLE UNKNOWN_SCHEMA." + TBL_NAME),
+            SQLException.class
+        );
+    }
+
+    /** */
+    @Ignore("https://ggsystems.atlassian.net/browse/GG-25387")
+    @Test
+    public void testDropIfExistsNonExistingSchema() {
+        sql("DROP TABLE IF EXISTS UNKNOWN_SCHEMA." + TBL_NAME);
+    }
+
+    /** */
+    private void executeStmtsAndVerify(Supplier<Boolean> withSchemaDecisionSup) {
+        sql("CREATE TABLE " + tableName(withSchemaDecisionSup.get()) + " (id INT PRIMARY KEY, val INT)");
+
+        sql("CREATE INDEX t1_idx_1 ON " + tableName(withSchemaDecisionSup.get()) + "(val)");
+
+        sql("INSERT INTO " + tableName(withSchemaDecisionSup.get()) + " (id, val) VALUES(1, 2)");
+        sql("SELECT * FROM " + tableName(withSchemaDecisionSup.get()), res -> oneRowList(1, 2).equals(res));
+
+        sql("UPDATE " + tableName(withSchemaDecisionSup.get()) + " SET val = 5");
+        sql("SELECT * FROM " + tableName(withSchemaDecisionSup.get()), res -> oneRowList(1, 5).equals(res));
+
+        sql("DELETE FROM " + tableName(withSchemaDecisionSup.get()) + " WHERE id = 1");
+        sql("SELECT COUNT(*) FROM " + tableName(withSchemaDecisionSup.get()), res -> oneRowList(0).equals(res));
+
+        sql("SELECT COUNT(*) FROM " + QueryUtils.SCHEMA_SYS + ".TABLES WHERE schema_name = 'PUBLIC' " +
+            "AND table_name = \'" + TBL_NAME + "\'", res -> oneRowList(1).equals(res));
+
+        sql("DROP TABLE " + tableName(withSchemaDecisionSup.get()));
+    }
+
+    /** */
+    private List<List<?>> oneRowList(Object... args) {
+        return Collections.singletonList(Arrays.asList(args));
+    }
+
+    /** */
+    protected void sql(String qry) {
+        sql(qry, null);
+    }
+
+    /** */
+    protected void sql(String qry, @Nullable Predicate<List<List<?>>> validator) {
+        List<List<?>> res = execSql(qry);
+
+        if (validator != null)
+            validator.test(res);
```

Comment:
Looks like validation result is ignored. Let's wrap it with `Assert.assertTrue()`. Also constants on rows 115 and 118 should be changed to long (0 -> 0L and 1 -> 1L accordingly).

Reply:
fixed